### PR TITLE
Add a new rule forbidding negative stop_time

### PR DESCRIPTION
**Pull request by stuebinm**
This adds an error if a `stop_time` has its `arrival_time` later than its `departure_time` (i.e. a vehicle would remain at the stop for a negative period of time), along with a corresponding test for this error.

As test data I have added a minimised version of the actual gtfs in which this issue just bit me ;)

(note that the MobilityData gtfs-validator has a more general check which also catches this issue: https://gtfs-validator.mobilitydata.org/rules.html#start_and_end_range_out_of_order-rule)

### CODE REVIEW

File: src/validators/stop_times.rs [hunk 1]
```diff
@@ -72,6 +73,31 @@ fn check_valid_stop_sequence(gtfs: &gtfs_structures::Gtfs) -> impl Iterator<Item
     })
 }
 
+fn check_times_increase(gtfs: &gtfs_structures::Gtfs) -> impl Iterator<Item = Issue> + '_ {
+    gtfs.trips
+        .values()
+        .map(|trip| {
+            trip.stop_times.iter().filter_map(move |st| {
+                match (st.arrival_time, st.departure_time) {
```
Comment (AntoineAugusti) [inline src/validators/stop_times.rs:81]: Wondering if this is safe to compare with this https://github.com/rust-transit/gtfs-structure/issues/122#issue-1172599876 in mind?

File: src/issues.rs [hunk 2]
```diff
@@ -30,6 +30,8 @@ pub enum IssueType {
     ExcessiveSpeed,
     /// The travel duration between two stops is negative.
     NegativeTravelTime,
+    /// A stop_time's departure is earlier than its arrival.
+    NegativeStopDuration,
```
Comment (AntoineAugusti) [inline src/issues.rs:34]: This should be added to the README

**Comment by AntoineAugusti** [comment]
@stuebinm Hi, thanks a lot for your contribution, much appreciated!

In which context do you use this library if you're willing to share?

File: src/validators/stop_times.rs (hunk 1 above)
Comment (stuebinm) [reply src/validators/stop_times.rs:81]: `std::time::Duration` [also implements][1] `Ord`, so I believe if the type of `arrival_time` & `departure_time` were changed this comparison would still work the same way (and if not the test will fail).

[1]: https://doc.rust-lang.org/std/time/struct.Duration.html#impl-Ord-for-Duration

File: src/issues.rs (hunk 2 above)
Comment (stuebinm) [reply src/issues.rs:34]: have added it

**Comment by stuebinm** [comment]
@antoine-de 

> In which context do you use this library if you're willing to share?

I use it for [Ilztalbahn](https://ilztalbahn.eu/) (a small volunteer-association operated railway line) as an automated check before publishing a new gtfs version (hence my earlier PR about making daemon support optional https://github.com/etalab/transport-validator/pull/197). Since we only have the one line and only a couple trains per week there's little automation in how the gtfs is generated, and sometimes people edit it by hand, so it's very useful to run this validator as a sanity check on it.

(btw, thanks a lot for this project! It's very cool to have open-source tools for handling gtfs which we can use, and sometimes even contribute back to!)

**Comment by stuebinm** [comment]
short note on the last force-push: I have added the new validation to the readme file as requested, and have also downgraded it from `Severity::Error` to `Severity::Warning` to be in line with e.g. `NullDuration` and `NegativeTravelTime` which are likewise warnings, not errors.

**Review by antoine-de — APPROVED** [review]
great, thanks!

**Comment by AntoineAugusti** [comment]
Will merge after adding the required error messages and translations on transport.data.gouv.fr

Thanks for your contribution @stuebinm, much appreciated 🙏🥳

**Pull request merged**